### ISSUE TITLE
[a11y]: VoiceOver not announcing toast notification when launched from a modal

### ISSUE THREAD

**Issue by kenneth-nguyen1** [carbon-design-system/carbon#21144]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

^1.92.1

### React version

18.2.0

### Automated testing tool and ruleset

AVT Audit with manual screen reader testing

### Assistive technology

MacOS VoiceOver

### Description

On Chrome browser, there is an accessibility issue with the `ToastNotification` component when used with a modal. When the toast is triggered from the modal’s primary button, VoiceOver does not announce the toast content. Instead, it announces the page `<title>` DOM element. If I trigger the same toast from a button on the page, VoiceOver announces it as expected. I attached a recording of this interaction below

https://github.com/user-attachments/assets/0ad32a50-11ef-4806-98b3-b9343bcaefa7

This recording example can be interacted with in this StackBlitz sandbox [here](https://stackblitz.com/edit/github-jgsfyz2w-pe3b4kin?file=src%2FApp.tsx. The only relevant change I made is setting `role="alert"` on the `ToastNotification` (instead of its default `role="status"`). This allowed the toast to announce correctly when triggered directly from the page with the button, but the issue still occurred when triggered from within a modal.

Is there a workaround to this or a possible fix for the Carbon component interactions?

### WCAG 2.1 Violation

4.1.3 Status Messages

### Reproduction/example

https://stackblitz.com/edit/github-jgsfyz2w-pe3b4kin?file=src%2FApp.tsx

### Steps to reproduce

On the Stackblitz sandbox:

1. Use screenreader tool to click on the "Open Modal" button
2. See modal open, and click on "Submit" primary button
3. See that the toast notification has appeared but the screenreader is not announcing it
4. (Optionally) Use the screenreader tool to click on the "Add toast" button and see that the screenreader is announcing the toast notification launched from that button

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
